python/_kastoremodule.c

```diff
@@ -284,19 +331,56 @@ static PyObject *
 kastore_dump(PyObject *self, PyObject *args, PyObject *kwds)
 {
     int err;
+    int fileobj_fd, new_fd;
     PyObject *ret = NULL;
     char *filename;
     kastore_t store;
     PyObject *data = NULL;
+    PyObject *fileobj = NULL;
+    PyObject *bytes = NULL;
+    FILE *file = NULL;
     static char *kwlist[] = {"data", "filename", NULL};
 
     memset(&store, 0, sizeof(store));
 
-    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O!s:dump", kwlist,
-                &PyDict_Type, &data, &filename)) {
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O!O:dump", kwlist,
+                &PyDict_Type, &data, &fileobj)) {
         goto out;
     }
-    err = kastore_open(&store, filename, "w", 0);
+    /* Try to obtain a file descriptor from fileobj. This succeeds if the
+     * fileobj has a fileno() method, such as objects derived from io.IOBase.
+     * But it also succeeds if fileobj is an integer file descriptor (PyLong).
+     * There's no reasonable use case for integer file descripter arguments
+     * here, and this is most likely user error. So we explicitly disallow
+     * PyLong fileobjs, to avoid unexpected behaviour (e.g. corrupting the data
+     * stream of any already open file).
+     */
+    fileobj_fd = PyObject_AsFileDescriptor(fileobj);
+    if (fileobj_fd != -1 && !PyLong_Check(fileobj)) {
```

Comment:
Might be safer to use ``PyNumber_Check``, as perhaps things like ``1.0`` would also succeed?

Reply:
I actually looked at the `PyObject_AsFileDescriptor` source code for this one. It does a PyLong_Check itself. Non-integer file descriptors certainly aren't supported in any OS I've heard of, so I don't think that PyObject_AsFileDescriptor will change that behaviour in a hurry.

Reply:
But probably adding a test that passes a float here is a good idea.